blob/ribs_store.go

```diff
@@ -0,0 +1,253 @@
+package blob
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/ipfs/boxo/chunker"
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	"github.com/lotus-web3/ribs"
+	"github.com/lotus-web3/ribs/rbdeal"
+	"github.com/multiformats/go-multihash"
+)
+
+// TODO parameterize this.
+const ribsStoreChunkSize = 4 << 20 // 4 MiB
```

Comment:
2MB is the max block size for a lot of protocols (Bitswap, etc). Silly, but probably we should cap at 1MB.